tests/e2e_tests/nonlin_e2e.py

```diff
@@ -164,7 +164,7 @@ def test_nonlin_cal_e2e(
 
     # Run the walker on some test_data
     print('Running walker')
-    walker.walk_corgidrp(nonlin_l1_list, '', e2eoutput_path)
+    walker.walk_corgidrp(nonlin_l1_list, '', e2eoutput_path, "l1_to_l2a_nonlin.json")
```

Comment:
I assume this is a temporary fix, since a number of things will change once Sergi finishes with the new filesorting thing, right?

Reply:
we need a new e2e test eventually to test walk_corgidrp() producing both a kgain and nonlin at the same time. I don't think it should be in this test, because the current kgain and nonlin e2e tests focus on checking consistency with TVAC. And I don't think mixing the two tests together is a good idea.

Reply:
Ok, sounds good. Should we create an issue for the new e2e test that we'll need, or is it implied that it should be included in Sergi's PR?

Reply:
new issue. we probably won't get to it today. I'm thinking it's less necessary since the functionality is there, but we don't need to have all the automation done this time around.

Reply:
works for me as long as the issue is created